morphir/runtime/src/org/finos/morphir/runtime/ToMDMConcept.scala

```diff
@@ -0,0 +1,165 @@
+package org.finos.morphir.runtime
+
+import org.finos.morphir.ir.conversion
+
+import org.finos.morphir.naming._
+import org.finos.morphir.datamodel.{Concept, Label, EnumLabel}
+import org.finos.morphir.ir.AccessControlled
+import org.finos.morphir.ir.{Type => T}
+import org.finos.morphir.ir.Type.{Type, UType}
+
+import scala.collection.immutable.{Map, Set}
+import org.finos.morphir.runtime.Distributions
+import org.finos.morphir.datamodel.Concept.Defaults
+import org.finos.morphir.runtime.Extractors.Types.*
+
+trait ToMDMConcept[A] { self =>
+  import ToMDMConcept.*
+  def apply(distributions: Distributions, bindings: Map[Name, Concept] = Map.empty): ToMDMConceptEither
+
+  final def concept(distributions: Distributions, bindings: Map[Name, Concept] = Map.empty): ToMDMConceptEither =
+    apply(distributions, bindings)
+
+  def as[B]: ToMDMConcept[B] = new ToMDMConcept[B] {
+    override def apply(distributions: Distributions, bindings: Map[Name, Concept] = Map.empty): ToMDMConceptEither =
+      self.apply(distributions, bindings)
+  }
+}
+
+object ToMDMConcept {
+
+  sealed trait NoEquivalentMDM         extends Throwable
```

Comment:
Would prefer to see this as an error hierarchy that starts with an error trait, i.e.:

```scala
sealed trait Error extends Exception
sealed trait NoEquivalentMDMRepr extends Error
// etc
```

By starting this with the `Error` it gives us a nice parachute for advancing the API.

This means that

```scala
type ToMDMConceptEither = Either[Error, Concept]
```